subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -323,24 +324,29 @@ private SnapshotHierarchy withWatcherChangeErrorHandling(SnapshotHierarchy curre
         try {
             return supplier.get();
         } catch (Exception ex) {
-            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD);
+            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD, null);
             return stopWatchingAndInvalidateHierarchyAfterError(currentRoot);
         }
     }
 
-    private void logWatchingError(Exception exception, String fileWatchingErrorMessage) {
-        if (exception instanceof InotifyInstanceLimitTooLowException) {
-            LOGGER.warn("{}. The inotify instance limit is too low. See {} for more details.",
-                fileWatchingErrorMessage,
-                daemonDocumentationIndex.getLinkToSection("sec:inotify_instances_limit")
-            );
-        } else if (exception instanceof InotifyWatchesLimitTooLowException) {
-            LOGGER.warn("{}. The inotify watches limit is too low.", fileWatchingErrorMessage);
-        } else if (exception instanceof WatchingNotSupportedException) {
-            // No stacktrace here, since this is a known shortcoming of our implementation
-            LOGGER.warn("{}. {}.", fileWatchingErrorMessage, exception.getMessage());
-        } else {
-            LOGGER.warn(fileWatchingErrorMessage, exception);
+    private void logWatchingError(Exception exception, String fileWatchingErrorMessage, @Nullable WatchMode watchMode) {
+        // Only log warnings when watching is explicitly enabled or on the info log.
+        // Since we don't know the watch mode during the build, we'll only log on info there.
```

Comment:
We aren't logging on `INFO`, though. We are logging on `WARN` but only when `INFO` is enabled. This will show up differently in the log files. I hate that we have to overcomplicate this instead of simply changing log levels...

Reply:
Yeah, it is very annoying. It would also be much easier if SLF4J would allow me to pass the log level as an argument. Note that this will only show up differently in logs if you log with `--debug`, because for info and warn we don't show the actual level in the output.

Do you have an idea to make it better?

Reply:
> Note that this will only show up differently in logs if you log with `--debug`, because for info and warn we don't show the actual level in the output.

For now that's true, though I maintain hope (maybe unreasonably) that at one point we'll have a properly configurable logging system.

Reply:
One workaround I found that I used in logging the build cache key building process is to pass in a `Logger` that is substituted with a no-op implementation if we shouldn't log. Not perfect, but it might be worth trying it out here to see if it helps.

We could also determine the logger instance to use at build start perhaps, and then reuse it until the next build. That would make passing `null` unnecessary, which is a plus I think.